util/src/Pos/Util/Wlog.hs

```diff
@@ -1,23 +1,288 @@
 
-import           System.Wlog (CanLog (..), HandlerWrap (..), HasLoggerName (..),
-                     LogEvent (..), LoggerConfig (..), LoggerName (..),
-                     LoggerNameBox (..), NamedPureLogger (..), Severity (..),
-                     WithLogger, consoleActionB, debugPlus,
-                     defaultHandleAction, dispatchEvents, errorPlus,
-                     fromScratch, hwFilePath, infoPlus, launchNamedPureLog,
-                     lcLogsDirectory, lcTermSeverityOut, lcTree, logDebug,
-                     logError, logInfo, logMCond, logMessage, logNotice,
-                     logWarning, ltFiles, ltSeverity, ltSubloggers,
+import           System.Wlog (HandlerWrap (..), HasLoggerName (..),
+                     LoggerConfig (..), LoggerName (..), LoggerNameBox (..),
+                     Severity (..), consoleActionB, debugPlus,
+                     defaultHandleAction, errorPlus, fromScratch, hwFilePath,
+                     infoPlus, lcLogsDirectory, lcTermSeverityOut, lcTree,
+                     logMCond, ltFiles, ltSeverity, ltSubloggers,
                      maybeLogsDirB, modifyLoggerName, noticePlus,
                      parseLoggerConfig, productionB, removeAllHandlers,
-                     retrieveLogContent, runNamedPureLog, setLevel,
-                     setupLogging, showTidB, termSeveritiesOutB,
-                     updateGlobalLogger, usingLoggerName, warningPlus,
-                     zoomLogger)
+                     retrieveLogContent, setLevel, setupLogging, showTidB,
+                     termSeveritiesOutB, updateGlobalLogger, usingLoggerName,
+                     warningPlus, zoomLogger)
 import           System.Wlog.Formatter (centiUtcTimeF)
 import           System.Wlog.LogHandler (LogHandlerTag (HandlerFilelike))
+
+--import qualified Pos.Util.Log as Log
+
+import           Universum
+
+import           Control.Monad.Morph (MFunctor (..))
+import qualified Control.Monad.State.Lazy as StateLazy
+import           Data.Sequence ((|>))
+import           Data.Text (Text)
+
+{-
+  attempt for reducing complexity:
+
+  1. count call sites per function
+  2. comment out 0 usages
+  3. group by "common usage pattern"
+      * just logging functions
+      * setup logging
+      * logging configuration
+      * pure logger?
+      * only 'CanLog' imported
+      * CanLog and WithLogger imported => redundant
+      * CanLog and HasLoggerName imported => replace with 'WithLogger'
+-}
+
+{- compatibility -}
+
+-- setupLogging will setup global shared logging state (MVar)
+-- usingLoggerName launches an action (of 'LoggerNameBox m a' ~ 'ReaderT LoggerName m a')
+-- LoggerNameBox is an instance of CanLog, HasLoggerName
+-- our functions are running in this context: 'type WithLogger m = (CanLog m, HasLoggerName m)'
+
+-- this fails with 'MonadIO' redundant constraint
+-- type WithLogger m = (CanLog m, HasLoggerName m, Log.LogContext m)
+
+{-
+class GHC.Base.Monad m => CanLog (m :: * -> *) where
```

Comment:
There's something that is obviously a comment above (compatibility), but is this a comment or commented out code? If its the latter, maybe we should remove it.